src/executables/master_heartbeat.c

```diff
@@ -6598,6 +6604,57 @@ hb_help_sprint_jobs_info (HB_JOB * jobs, char *buffer, int max_length)
   return p - buffer;
 }
 
+/**
+ * Compare two host names if are equal, if one of the host names is canonical name and the other is not, then
+ * only host part (e.g. for canonical name "host-1.cubrid.org" host part is "host-1") is used for comparison
+ *
+ * for example following hosts are equal:
+ *  "host-1"            "host-1"
+ *  "host-1"            "host-1.cubrid.org"
+ *  "host-1.cubrid.org" "host-1"
+ *  "host-1.cubrid.org" "host-1.cubrid.org"
+ *
+ * for example following hosts are not equal:
+ *  "host-1"            "host-2"
+ *  "host-1.cubrid.org" "host-2"
+ *  "host-1"            "host-2.cubrid.org"
+ *  "host-1.cubrid.org" "host-2.cubrid.org"
+ *  "host-1.cubrid.org" "host-1.cubrid.com"
+ *
+ * @param hostname_a first hostname
+ * @param hostname_b second hostname
+ *
+ * @return true if hostname_a is same as hostname_b
+ */
+static bool
+are_hostnames_equal (const char *hostname_a, const char *hostname_b)
```

Comment:
I think the function is good and we may save `strlen` as follows:

```c
  const char *a;
  const char *b;

  for (a = hostname_a, b = hostname_b; *a && *b && (*a == *b); a++, b++)
    ;

  if (*a == '\0' && *b != '\0')
    {
      // if a reached the end and b does not, b must be '.'
      return *b == '.';
    }
  else if (*a != '\0' && *b == '\0')
    {
      // if b reached the end and a does not, a must be '.'
      return *a == '.';
    }
  else
    {
      return *a == *b;
    }
```

Since this is a tiny optimization and clarity is more important, you can go with your preference :)